lib/dawdle/message_encoder/message_encoder.ex

```diff
@@ -16,4 +16,38 @@ defmodule Dawdle.MessageEncoder do
   Decode a string pulled from the queue into its original representation.
   """
   @callback decode(message :: String.t()) :: any()
+
+  @default_encoder Dawdle.MessageEncoder.Term
+
+  @spec encode(any()) :: String.t()
+  def encode(event), do: timed_op(:encode, event)
+
+  @spec decode(String.t()) :: any()
+  def decode(message), do: timed_op(:decode, message)
+
+  defp get_encoder do
+    Application.get_env(:dawdle, :encoder, @default_encoder)
+  end
+
+  defp timed_op(op, data) do
```

Comment:
It's not critical, but it seems like maybe this function could be generalised out to be used in all the other timing operations to cut down on the repeated code structure used for timing. Allow it to take a fun and parameters to fill out - it would also reduce the risk of typos causing mis-matched begin/end blocks.

Reply:
I was originally really bothered by the duplicate code for the measurements, but the function that I came up with to encapsulate that logic was worse. After sleeping on it, I have come up with a more elegant solution.